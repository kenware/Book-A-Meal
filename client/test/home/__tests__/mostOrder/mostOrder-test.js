import React from 'react';
import { shallow } from 'enzyme';
import MostOrder from '../../../../components/home/mostOrder';
import { props, emptyProps } from '../../homeMockData';

describe('Test mostOder components', () => {
  it('renders correctly', () => {
    const tree = shallow(<MostOrder {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should have one meal-menu-row', () => {
    const tree = shallow(<MostOrder {...props} />);
    expect(tree.find('.meal-menu')).toHaveLength(1);
  });

  it('should have one meal-menu', () => {
    const tree = shallow(<MostOrder {...props} />);
    expect(tree.find('.meal-menu').length).toEqual(1);
  });

  it('should have no meal-menu with empty props', () => {
    const tree = shallow(<MostOrder {...emptyProps} />);
    expect(tree.find('.meal-menu').length).toEqual(0);
  });

  it('should have none div', () => {
    const tree = shallow(<MostOrder {...props} />);
    expect(tree.find('div').length).toEqual(6);
  });

  it('should have none link', () => {
    const tree = shallow(<MostOrder {...props} />);
    expect(tree.find('Link').length).toEqual(1);
  });

  it('it should flip card on mouseEnter event', () => {
    let tree = shallow(<MostOrder {...props} />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<MostOrder {...emptyProps} />);
    expect(tree).toMatchSnapshot();
  });
});
