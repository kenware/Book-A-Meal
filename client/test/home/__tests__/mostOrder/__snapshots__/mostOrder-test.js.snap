// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test mostOder components it should flip card on mouseEnter event 1`] = `
<div
  className="meal-menu-row"
>
  <div
    className="meal-menu"
    key="1"
  >
    <div
      className="meal-menu-card"
    >
      <div
        className="meal-img-wrap"
      >
        <div
          className="meal-overlay"
        />
        <img
          alt="Avatar"
          className="meal-card-img"
          src="image"
        />
      </div>
      <div
        className="container l-r-pad-text"
      >
        <h4
          className="p-color"
        >
          <b>
            $
            400
          </b>
        </h4>
        <p>
          rice
        </p>
        <Link
          replace={false}
          to="/detail/1"
        >
          View
        </Link>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test mostOder components it should flip card on mouseEnter event 2`] = `
<div
  className="meal-menu-row"
/>
`;

exports[`Test mostOder components renders correctly 1`] = `
<div
  className="meal-menu-row"
>
  <div
    className="meal-menu"
    key="1"
  >
    <div
      className="meal-menu-card"
    >
      <div
        className="meal-img-wrap"
      >
        <div
          className="meal-overlay"
        />
        <img
          alt="Avatar"
          className="meal-card-img"
          src="image"
        />
      </div>
      <div
        className="container l-r-pad-text"
      >
        <h4
          className="p-color"
        >
          <b>
            $
            400
          </b>
        </h4>
        <p>
          rice
        </p>
        <Link
          replace={false}
          to="/detail/1"
        >
          View
        </Link>
      </div>
    </div>
  </div>
</div>
`;
